lib/libzfs/libzfs_sendrecv.c

```diff
@@ -941,28 +942,39 @@ send_iterate_fs(zfs_handle_t *zhp, void *arg)
 	}
 
 	if (nv != NULL)
-		VERIFY(0 == nvlist_add_nvlist(nvfs, "props", nv));
+		fnvlist_add_nvlist(nvfs, "props", nv);
 
 	/* iterate over snaps, and set sd->parent_fromsnap_guid */
 	sd->parent_fromsnap_guid = 0;
-	VERIFY(0 == nvlist_alloc(&sd->parent_snaps, NV_UNIQUE_NAME, 0));
-	VERIFY(0 == nvlist_alloc(&sd->snapprops, NV_UNIQUE_NAME, 0));
+	sd->parent_snaps = fnvlist_alloc();
+	sd->snapprops = fnvlist_alloc();
+	if (!sd->replicate && fromsnap_txg != 0) {
+		range_nvl = fnvlist_alloc();
+		fnvlist_add_uint64(range_nvl, SNAP_ITER_SKIP_TO_TXG,
+		    fromsnap_txg_save);
```

Comment:
Why does the check use `fromsnap_txg`, but we "skip to" `fromsnap_txg_save`?  From reading the code, it seems like `fromsnap_txg` is the txg of the fromsnap in this fs, whereas `fromsnap_txg_save` is the txg of the fromsnap in the parent filesystem.  So we should be using `fromsnap_txg` in both places here?  The test case would be one where the parent and child fromsnap's have different txg's (i.e. snapshots with the same name were created at different times).